javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -880,6 +880,50 @@ void setCurrentArray() {
         };
     }
 
+    /* TODO unify te two iterators somehow */
```

Comment:
... as the TODO states ...
This method will help with the primitivization iteration also, avoiding boxing